ast/conflicts.go

```diff
@@ -27,7 +27,12 @@ func CheckPathConflicts(c *Compiler, exists func([]string) (bool, error)) Errors
 
 func checkDocumentConflicts(node *TreeNode, exists func([]string) (bool, error), path []string) Errors {
 
-	path = append(path, string(node.Key.(String)))
+	switch key := node.Key.(type) {
+	case String:
+		path = append(path, string(key))
+	default: // other key types cannot conflict with data
+		return nil
+	}
```

Comment:
Technically, there could be conflicts between data and non-string object keys, like
```
package x
p.q.r[3] = true if true
```
and `data.x.p.q.r =  {"3": false }`, but that's because we need to string-ify non-string keys at some point. However, we've accepted that clash before.